### ISSUE TITLE
Spyre coarse tiling mis-lowers unsqueeze consumer of amax result, causing incorrect denominator

### ISSUE THREAD

**Issue by cyang49** [torch-spyre/torch-spyre#3003]
### What happened?

A FlashAttention-like coarse-tiling case produces incorrect results when an `amax(..., dim=-1)` result is consumed through `unsqueeze(-1)` inside the same coarse-tiled fused kernel.

The issue appears to be in the generated layout for the `block_max.unsqueeze(-1)` input to a later `sub`. The `amax` output itself appears correct, but the subsequent read uses a device layout with an extra dimension that changes the physical row stride.

**Minimal Repro**
Script:

```text
test-spyre-scripts/repro_coarse_unsqueeze_denominator.py
```

Core compiled function:

```python
def denominator(q, k, mask):
    with spyre_hint(tiles={"B": 1}):
        with spyre_hint(tiles={"H": 1}):
            with spyre_hint(tiles={"Lq": 2}):
                with spyre_hint(tiles={"Lk": 1}):
                    with spyre_hint(work_div={"H": 4, "Lq": 8, "Lk": 8}):
                        scores = torch.matmul(q, k.transpose(-1, -2)) + mask
                        block_max = torch.amax(scores, dim=-1)
                        exp_scores = torch.exp(scores - block_max.unsqueeze(-1))
                        return exp_scores.sum(dim=-1)
```

The mask is all zeros, so it should not affect the math. It is included because it preserves the problematic fused/coarse-tiled graph shape.

**How To Run**
```bash
rm -rf /tmp/spyre_clean_repro_cache /tmp/spyre_clean_repro_debug

TORCHINDUCTOR_CACHE_DIR=/tmp/spyre_clean_repro_cache \
TORCH_COMPILE_DEBUG=1 \
TORCH_COMPILE_DEBUG_DIR=/tmp/spyre_clean_repro_debug \
SPYRE_INDUCTOR_LOG=1 \
SPYRE_INDUCTOR_LOG_LEVEL=debug \
python test-spyre-scripts/repro_coarse_unsqueeze_denominator.py
```

**Observed Result**
Example output:

```text
mismatches=3114/4096 max_abs=653.0
worst_index=(0, 3, 382) expected=289.75 actual=943.0
```

The generated logs show coarse tiling is active:

```text
coarse_tile_groups: 1 group(s) formed
LoopSpec(count=2)
```

**What To Look For**
Inspect the generated wrapper:

```bash
rg -n "device_size=\\[1, 512, 4, 2, 64\\]|LoopSpec|op='max'|op='sub'" \
  /tmp/spyre_clean_repro_cache -g '*.py'
```

Suspicious generated layout:

```python
# amax writes block_max with the expected tile layout:
device_size=[1, 512, 4, 64]
device_coordinates=[0, c1, c0, 0]

# later sub reads block_max.unsqueeze(-1) with an extra physical dimension:
device_size=[1, 512, 4, 2, 64]
device_coordinates=[0, c1, c0, 0, 0]
```

That extra `2` changes the physical stride of `c1`, so the `sub` appears to read the wrong row of `block_max`.

**Expected Result**
The compiled Spyre result should match the CPU reference within tolerance. The `unsqueeze(-1)` consumer of the `amax` result should preserve a layout compatible with the original `amax` output, rather than introducing a physical dimension that changes row addressing.

### What did you expect to happen?

the results should match

### How can we reproduce it?

Use this reproducer script
[repro_coarse_unsqueeze_denominator.py](https://github.com/user-attachments/files/29573009/repro_coarse_unsqueeze_denominator.py)

### Any environmental details we need to know?

No

### Relevant log output

```Shell
mismatches=3114/4096 max_abs=653.0
worst_index=(0, 3, 382) expected=289.75 actual=943.0
```

**Comment by cyang49** [comment]
Summary with experiments:

I compared two `test_flash.py` configurations.

**Case A: `Lq=4096`, `q_block_size=1024`**

This generated the current wrapper:

```text
/tmp/torchinductor_ccyang/3t/c3trdqs7hrazuhmdsfcv6asmp64fp34m4pa6zkzgmaiieig6cowf.py
```

The wrapper still shows the suspicious layout pattern: sparse `[B,H,Lq]` values produced as `[1,1024,4,64]` are later consumed through unsqueezed/broadcast views like `[1,1024,4,4,64]`.

But the final output matched closely:

```text
max_abs 0.01123046875
mean_abs 0.0001497
bad at 0.01/0.01: 0 / 16777216
bad at 0.1/0.1:  0 / 16777216
```

So the bug pattern appears present, but is mostly hidden in final attention output.

**Case B: `Lq=1024`, `q_block_size=512`**

This generated:

```text
/tmp/torchinductor_ccyang/vc/cvc3yhuqwxdopk4bkim45njifhxahvtsdlscaupgx3i4wx43w5gi.py
```

Here the wrapper shows the cleaner suspicious pattern: `max`/`exp` produce `[1,512,4,64]`, while unsqueezed consumers read `[1,512,4,2,64]`.

The final output failed badly:

```text
max_abs 3.173828125
mean_abs 0.0612
bad at 0.1/0.1: 753502 / 4194304
worst: expected -2.59765625, actual 0.576171875
```

**Theory**

With only one `Lk` tile, a wrong row-wise max can sometimes cancel in:

```text
exp(scores - m) @ V / sum(exp(scores - m))
```

That likely explains why the `Lq=4096` case passes despite suspicious artifacts.

For `Lq=1024, q_block_size=512`, the wrong max/read offset seems large enough to break fp16 softmax stability. Then the numerator and denominator no longer cancel, exposing the same underlying layout/view bug as a visible correctness failure.
